src/front-door/azext_front_door/_params.py

```diff
@@ -263,3 +266,59 @@ def load_arguments(self, _):
         c.argument('rule_name', options_list=['--name', '-n'], help='Name of the custom rule.')
         c.argument('policy_name', waf_policy_name_type, id_part=None)
     # endregion
+
+    # region RulesEngine
+    with self.argument_context('network front-door rules-engine rule') as c:
+        c.argument('front_door_name', frontdoor_name_type, options_list=['--front-door-name', '-n'])
+        c.argument('rules_engine_name', rules_engine_name_type, id_part=None)
+        c.argument('rule_name', help='Name of the rule')
+        c.argument('action_type', arg_group="Action", arg_type=get_enum_type(['RequestHeader', 'ResponseHeader', 'ForwardRouteOverride', 'RedirectRouteOverride']), help='Action type to apply for a rule.')
+        c.argument('header_action', arg_group="Action", arg_type=get_enum_type(HeaderActionType), help='Header action type for the requests.')
+        c.argument('header_name', arg_group="Action", help='Name of the header to modify.')
+        c.argument('header_value', arg_group="Action", help='Value of the header.')
+        c.argument('match_variable', arg_group="Match Condition", arg_type=get_enum_type(RulesEngineMatchVariable), help='Name of the match condition.')
+        c.argument('operator', arg_group="Match Condition", arg_type=get_enum_type(RulesEngineOperator), help='Operator of the match condition.')
+        c.argument('match_values', arg_group="Match Condition", nargs='+', help='Space-separated list of values to match against.')
+        c.argument('selector', arg_group="Match Condition", help='Optional selector for the match condition variable.')
```

Comment:
should it be an enum? what are the valid values?

Reply:
No, selector is a [string](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/frontdoor/resource-manager/Microsoft.Network/stable/2020-01-01/frontdoor.json#L1921)

Reply:
got it